evaluate.go

```diff
@@ -237,10 +237,37 @@ func evaluateNotPresent(ptr pointerstructure.Pointer, datum interface{}) bool {
 	return reflect.ValueOf(val).Kind() == reflect.Map
 }
 
-func evaluateMatchExpression(expression *grammar.MatchExpression, datum interface{}, opt ...Option) (bool, error) {
+func getValue(datum interface{}, path []string, opt ...Option) (interface{}, bool, error) {
 	opts := getOpts(opt...)
+	if len(path) != 0 {
```

Comment:
Style nit -- and this is personal style so feel free to leave as-is. Even though the for loop below won't run if there are no local variables, I think it's cleaner for the casual reader to specify this as `if len(path) != 0 && len(opts.withLocalVariables) > 0`. It just makes it clear that the _only_ function of the code in this `if` branch is if both conditions are true.